src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonInputFormatter.cs

```diff
@@ -98,5 +131,68 @@ private Stream GetInputStream(HttpContext httpContext, Encoding encoding)
 
             return new TranscodingReadStream(httpContext.Request.Body, encoding);
         }
+
+        // Keep in sync with NewtonsoftJsonInputFormatter.GetPathMetadata
+        private ModelMetadata GetPathMetadata(ModelMetadata metadata, string path)
+        {
+            var index = 0;
+            while (index >= 0 && index < path.Length)
+            {
+                if (path[index] == '[')
+                {
+                    // At start of "[0]".
+                    if (metadata.ElementMetadata == null)
+                    {
+                        // Odd case but don't throw just because ErrorContext had an odd-looking path.
+                        break;
+                    }
+
+                    metadata = metadata.ElementMetadata;
+                    index = path.IndexOf(']', index);
+                }
+                else if (path[index] == '.' || path[index] == ']')
+                {
+                    // Skip '.' in "prefix.property" or "[0].property" or ']' in "[0]".
+                    index++;
+                }
+                else
+                {
+                    // At start of "property", "property." or "property[0]".
+                    var endIndex = path.IndexOfAny(new[] { '.', '[' }, index);
+                    if (endIndex == -1)
+                    {
+                        endIndex = path.Length;
+                    }
+
+                    var propertyName = path.Substring(index, endIndex - index);
+                    if (metadata.Properties[propertyName] == null)
+                    {
+                        // Odd case but don't throw just because ErrorContext had an odd-looking path.
+                        break;
+                    }
+
+                    metadata = metadata.Properties[propertyName];
+                    index = endIndex;
+                }
+            }
+
+            return metadata;
+        }
+
+        private static class Log
+        {
+            private static readonly Action<ILogger, Exception> _jsonInputFormatterException;
+
+            static Log()
+            {
+                _jsonInputFormatterException = LoggerMessage.Define(
+                   LogLevel.Debug,
+                   new EventId(1, "SystemTextJsonInputException"),
+                   "JSON input formatter threw an exception.");
```

Comment:
Should we log more details here? What type of object failed to deserialize? Or the actual error message?

Also, should we log success as well?

Reply:
This is pretty much what we do with Json.NET. If we were to log a success message, a better place would be in `BodyModelBinder`: https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/ModelBinding/Binders/BodyModelBinder.cs#L157-L183

Reply:
I'm not making the argument that we have good logging elsewhere, I'm making the argument that we should have good logging everywhere.